File: core/core-api/src/main/java/com/ssafy/witch/controller/snack/response/SnackResponse.java
package com.ssafy.witch.controller.snack.response;

import com.ssafy.witch.controller.user.response.UserResponse;
import com.ssafy.witch.user.model.UserBasicProjection;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public class SnackResponse {

  private String snackId;
  private Double longitude;
  private Double latitude;
  private String snackImageUrl;
  private String snackSoundUrl;
  private LocalDateTime createdAt;

  private UserResponse user;
}

File: Witch/app/src/main/java/com/ssafy/witch/data/model/response/SnackResponse.kt
package com.ssafy.witch.data.model.response

import com.google.gson.annotations.SerializedName

data class SnackResponse(
    @SerializedName("snacks") var snacks: List<SnackInfo>
){
    data class SnackInfo(
        val snackId: String,
        val longitude: Double,
        val latitude: Double,
        val snackImageUrl: String,
        val snackSoundUrl: String,
        val createdAt: String,
        @SerializedName("user") var user: UserInfo
    ) {
        data class UserInfo(
            val userId: String,
            val nickname: String,
            val profileImageUrl: String
        )
    }
}
